tests/integrations/pd/test_rpc_client.rs

```diff
@@ -218,6 +218,30 @@ fn test_validate_endpoints() {
     assert!(block_on(validate_endpoints(env, &new_config(eps), mgr)).is_err());
 }
 
+#[test]
+fn test_validate_endpoints_retry() {
+    let eps_count = 3;
+    let server = MockServer::with_case(eps_count, Arc::new(Split::new()));
+    let env = Arc::new(
+        EnvBuilder::new()
+            .cq_count(1)
+            .name_prefix(thd_name!("test-pd"))
+            .build(),
+    );
+    let eps = server.bind_addrs();
+    let mut copy_eps = eps.clone();
```

Comment:
Seems we do not need to clone a new vector here? (just iterate over the reference of the vector)